### PR TITLE
Added the Cauchy, Lognormal and Pareto thick-tailed distributions.

### PR DESCRIPTION
**Pull request by bombaywalla**
Added tests.
The Cauchy is missing a converges-to-mean? test since its mean is not
defined.

Updated README.

### CODE REVIEW

**Comment by bombaywalla** [comment]
Not sure whether you'd prefer lognormal or log-normal in the API
(and correspondingly Lognormal or LogNormal in the deftypes).

File: src/kixi/stats/distribution.cljc [hunk 1]
```diff
@@ -506,6 +506,59 @@
...
+    (+ 0.5 (/ (atan (/ (- x location) scale)) PI)))
+  (quantile [this p]
+    (+ location (* scale (tan (* PI (- p 0.5))))))
+  #?@(:clj (clojure.lang.Seqable
+            (seq [this] (sampleable->seq this)))
+      :cljs (ISeqable
+             (-seq [this] (sampleable->seq this)))))
+
+(deftype ^:no-doc Lognormal
+  [logmu logsd]
```
Comment (henrygarner) [inline src/kixi/stats/distribution.cljc:527]: I'd prefer to have `mu` and `sd` parameter names here

File: src/kixi/stats/distribution.cljc [hunk 2]
```diff
@@ -506,6 +506,59 @@
...
+  (cdf [this x]
+    (+ 0.5 (/ (atan (/ (- x location) scale)) PI)))
+  (quantile [this p]
+    (+ location (* scale (tan (* PI (- p 0.5))))))
+  #?@(:clj (clojure.lang.Seqable
+            (seq [this] (sampleable->seq this)))
+      :cljs (ISeqable
+             (-seq [this] (sampleable->seq this)))))
+
+(deftype ^:no-doc Lognormal
```
Comment (henrygarner) [inline src/kixi/stats/distribution.cljc:526]: You guessed it, I'd prefer `LogNormal`

File: src/kixi/stats/distribution.cljc [hunk 3]
```diff
@@ -646,6 +699,27 @@
...
   (->DirichletMultinomial n alphas))
 
+(defn cauchy
+  "Returns a Cauchy distribution.
+  Params: {:location ∈ ℝ, :scale ∈ ℝ > 0}"
+  [{:keys [location scale]}]
+  (assert (pos? scale) (str "Scale (" scale ") must be positive"))
+  (->Cauchy location scale))
+
+(defn lognormal
```
Comment (henrygarner) [inline src/kixi/stats/distribution.cljc:709]: You guessed it,  I'd prefer `log-normal`

File: src/kixi/stats/distribution.cljc [hunk 4]
```diff
@@ -646,6 +699,27 @@
...
+  Params: {:location ∈ ℝ, :scale ∈ ℝ}"
+  [{:keys [location scale logmu logsd]}]
+  (->Lognormal (or location logmu) (or scale logsd)))
+
+(defn pareto
+  "Returns a Pareto distribution.
+  Params: {:scale ∈ ℝ > 0, :shape ∈ ℝ > 0}"
+  [{:keys [scale shape]}]
+  (assert (and (pos? scale) (pos? shape))
+          (str "Scale (" scale ") and shape (" shape "must be positive."))
```
Comment (henrygarner) [inline src/kixi/stats/distribution.cljc:720]: Unbalanced parens in the assert error string ( 😱

File: src/kixi/stats/distribution.cljc [hunk 5]
```diff
@@ -506,6 +506,59 @@
...
+  (sample-1 [this rng]
+    (exp (+ (* (rand-normal rng) logsd) logmu)))
+  (sample-n [this n rng]
+    (default-sample-n this n rng))
+  p/PQuantile
+  (cdf [this x]
+    (* 0.5 (+ 1 (erf (/ (- (log x) logmu)
+                           (sqrt (* 2 logsd logsd)))))))
+  (quantile [this p]
+    (exp (+ (* -1.41421356237309505 logsd (erfcinv (* 2 p))) logmu)))
```
Comment (henrygarner) [inline src/kixi/stats/distribution.cljc:538]: Checks out 👍

File: README.md [hunk 6]
```diff
@@ -9,14 +9,17 @@ A Clojure/ClojureScript library of statistical sampling and transducing function
...
 * Binomial
 * Categorical
+* Cauchy
 * Chi-Squared
 * Dirichlet
 * Dirichlet-multinomial
 * Exponential
 * F
 * Gamma
+* Lognormal
```
Comment (henrygarner) [inline README.md:19]: You guessed it, I'd prefer Log-normal here.

File: test/kixi/stats/distribution_test.cljc [hunk 7]
```diff
@@ -255,6 +267,10 @@
                             (sut/binomial {:n n :p p})))
     (is (converges-to-mean? a
                             (sut/normal {:location a :scale (/ 1 r)})))
+    (is (converges-to-mean? (exp (+ a (* 0.5 b b)))
+                            (sut/lognormal {:location a :scale b})))
+    (is (converges-to-mean? (/ (* s (+ (/ 1 r) 0.01)) (- s 1)) ; :scale > 1
```
Comment (henrygarner) [inline test/kixi/stats/distribution_test.cljc:272]: Why `:scale > 1` when 0 < x <= 1 is valid? 🤔

File: src/kixi/stats/distribution.cljc (hunk 1 above)
Comment (bombaywalla) [reply src/kixi/stats/distribution.cljc:527]: The reason I used `:logmu` and `:logsd` was the following text

> These are the expected value (or mean) and standard deviation of the variable's natural logarithm, not the expectation and standard deviation of {\displaystyle X}X itself.


on this [Wikipedia entry](https://en.wikipedia.org/wiki/Log-normal_distribution#Generation_and_parameters).

Just to make it clear that the parameters are not the same as a Normal distribution.

File: src/kixi/stats/distribution.cljc (hunk 2 above)
Comment (bombaywalla) [reply src/kixi/stats/distribution.cljc:526]: Okay. I will make the required changes.

File: src/kixi/stats/distribution.cljc (hunk 4 above)
Comment (bombaywalla) [reply src/kixi/stats/distribution.cljc:720]: Ah yes.  Thanks for catching that.

File: src/kixi/stats/distribution.cljc (hunk 5 above)
Comment (bombaywalla) [reply src/kixi/stats/distribution.cljc:538]: I was wondering whether having a set of tests that checked for

> (cdf distrib (quantile distrib x)) = x

would be worthwhile.

File: README.md (hunk 6 above)
Comment (bombaywalla) [reply README.md:19]: Ok. Will change.

File: test/kixi/stats/distribution_test.cljc (hunk 7 above)
Comment (bombaywalla) [reply test/kixi/stats/distribution_test.cljc:272]: Thanks for the catch.  It should be `:shape > 1` for the mean to be finite.
I will fix the code.

File: src/kixi/stats/distribution.cljc (hunk 1 above)
Comment (henrygarner) [reply src/kixi/stats/distribution.cljc:527]: The parameters are conventionally denoted as 𝜇 / mu and σ2 / sd, so I find those the least surprising way to parameterise the distribution (and the helper `log-normal` function), even though I follow your reasoning. I don't see a contradiction: they are the same parameters as the normal distribution, but in a different context, much like the shape, scale and rate parameters for other distributions.

Comment (bombaywalla) [reply src/kixi/stats/distribution.cljc:527]: Okay.  I'll change the names and add to the docstring to specify what the parameters refer to.

File: src/kixi/stats/distribution.cljc (hunk 5 above)
Comment (henrygarner) [reply src/kixi/stats/distribution.cljc]: Definitely, I really like that idea

**Pull request merged**